### PR TITLE
DEV: Refactor + fix small issues

### PR DESCRIPTION
**Pull request by jordanvidrine**
This PR refactors some of the logic behind when the docs-card-filter is shown to users as there was some faulty behavior when sorting the filtered topic list. I have added logic to *not* show a category card if it is currently selected as a filter to prevent redundancy. I also have added styling to accommodate for a large amount of category or tag card filters.

### CODE REVIEW

File: scss/card-filters.scss [hunk 1]
```diff
@@ -64,11 +114,14 @@
...
   }
   &-description {
-    font-size: $font-down-1;
-    display: -webkit-box;
-    -webkit-line-clamp: 2;
-    -webkit-box-orient: vertical;  
-    overflow: hidden;
+    margin: 0;
+    p {
+      font-size: $font-down-1;
```
Comment (CvX) [inline scss/card-filters.scss:119]: While here, could you convert these to `var()` syntax?

File: scss/card-filters.scss [hunk 2]
```diff
@@ -64,11 +114,14 @@
...
+      display: -webkit-box;
+      -webkit-line-clamp: 2;
+      -webkit-box-orient: vertical;
```
Comment (CvX) [inline scss/card-filters.scss:122]: Also, that's some outdated code! 😅 We have flex available for some time now!

File: javascripts/discourse/components/cards-filter.js [hunk 3]
```diff
@@ -14,45 +14,55 @@ export default Component.extend({
...
-        return pluginTags.indexOf(`${category.id}`) !== -1;
+      includedTags = tags.filter(tag => {
+        let currentTags = [];
+
+        if (params?.tags) {
+          currentTags.push(...params.tags.split("|"));
+        }
+
+        return (
+          pluginTags.indexOf(`${tag.id}`) !== -1 &&
```
Comment (CvX) [inline javascripts/discourse/components/cards-filter.js:63]:
```suggestion
          pluginTags.includes(`${tag.id}`) &&
```

File: javascripts/discourse/components/cards-filter.js [hunk 4]
```diff
@@ -14,45 +14,55 @@ export default Component.extend({
...
+      includedCategories = categories.filter(category => {
+        let currentCategory;
+
+        if (params?.category) {
+          currentCategory =
+            Number(params.category) === category.id ? category.id : "";
+        }
+
+        return (
+          pluginCategories.indexOf(`${category.id}`) !== -1 &&
```
Comment (CvX) [inline javascripts/discourse/components/cards-filter.js:40]:
```suggestion
          pluginCategories.includes(`${category.id}`) &&
```

File: javascripts/discourse/components/cards-filter.js [hunk 5]
```diff
@@ -14,45 +14,55 @@ export default Component.extend({
...
-            return false;
-          } else {
-            return true;
-          }
-        }
-      }
+    if (this?.includedCategories?.length > 0 || this?.includedTags?.length > 0) {
+      return true;
+    } else {
+      return false;
```
Comment (CvX) [inline javascripts/discourse/components/cards-filter.js:20]:
```suggestion
    return this.includedCategories?.length > 0 || this.includedTags?.length > 0;
```

File: scss/card-filters.scss (hunk 2 above)
Comment (jordanvidrine) [reply scss/card-filters.scss:122]: I believe this is required for line-camp to work properly.

Comment (CvX) [reply scss/card-filters.scss:122]: Ah, I see. We're close to `line-clamp` being widely supported, but not quite there yet. https://caniuse.com/?search=line-clamp

**Review by CvX — APPROVED** [review]

**Pull request merged**